core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -2725,9 +2725,11 @@ struct ViewDataHandle<
                                    Kokkos::CudaUVMSpace>::value))
 #endif
                 && (!Traits::memory_traits::is_atomic))>::type> {
-  using value_type  = typename Traits::value_type;
-  using handle_type = typename Traits::value_type* KOKKOS_IMPL_ALIGN_PTR(
-      KOKKOS_MEMORY_ALIGNMENT);
+  using value_type = typename Traits::value_type;
+  // typedef work-around for intel compilers error #3186: expected typedef
+  // declaration
+  typedef value_type* KOKKOS_IMPL_ALIGN_PTR(KOKKOS_MEMORY_ALIGNMENT)
+      handle_type;  // NOLINT(modernize-use-using)
```

Comment:
Does this the pass clang-tidy check? We might have to use `NOLINTNEXTLINE` or something if not

Reply:
Just pushed the `NOLINTNEXTLINE` change